<launch>
 <!-- <arg name="odom_topic" default ="/gazebo/forklift_controllers/odom"/> -->
 <arg name="global_frame" default ="odom"/>
 <arg name="path_frame" default ="base_link_p"/>
 <arg name="cmd_vel_topic" default ="/cmd_vel_raw"/>
 <arg name="pallet_topic" default ="/pallet_detection_relay2/pallet_pose"/>

 <node pkg="pallet_docking_xsquare" type="docking_planner" name="docking_planner" output = "screen" >
   <param name="global_frame" value="$(arg global_frame)"/>
   <param name="path_frame" value="$(arg path_frame)"/>

  <!-- Goal parameters -->
   <param name="approaching_min_dis" value="1.0"/>  
   <param name="dis_approach_offset" value="0.8"/>  
   <param name="dis_docking_offset" value="-0.57"/> <!-- -0.6 -->
   <param name="moveback_straight_distance" value="1.0"/>

  <!-- Starting boundary -->
   <param name="starting_vel" value="0.2"/>
   <param name="starting_acc" value="0.0"/>

  <!-- Ending boundary -->
   <param name="stopping_vel" value="0.2"/>
   <param name="stopping_acc" value="0"/>
   
  <!-- Max boundary -->
   <param name="max_yaw_rate" value="2.5"/>
   <param name="max_path_curvature" value="100.0"/>
   <param name="max_ax" value="1.0"/>
   <param name="max_ay" value="0.2"/>
   <param name="max_acc" value="0.5"/>
   <param name="max_jerk" value="0.2"/>

  <!-- Time condition -->
   <param name="min_t" value="1.0"/>
   <param name="max_t" value="10.0"/>
   <param name="dt" value="0.1"/>

  <!-- Goal tolerance -->
   <param name="distance_tolerance" value="0.1"/>  <!-- meter -->
   <param name="angle_tolerance" value="0.02"/>     <!-- radian -->  
   <param name="final_angle_tolerance" value="0.03"/>     <!-- radian -->  
   <param name="x_tolerance" value="0.035"/> <!-- meter -->    
   <param name="y_tolerance" value="0.03"/> <!-- meter -->     

  <!-- PID control for sideshift control -->
    <param name="kp" value="12.0"/>
    <param name="ki" value="2.5"/>
    <param name="kd" value="0.0"/>
    <param name="error_tolerance_" value="0.005"/>

    <remap from="/cmd_vel" to = "$(arg cmd_vel_topic)"/>
    <remap from="/pallet_detection_relay/pallet_pose" to = "$(arg pallet_topic)"/>
 </node>

</launch>
